### ISSUE TITLE
Can add rich rules when lockdown on

### ISSUE THREAD

**Issue by winnermy** [firewalld/firewalld#274]
I tested at firewall-cmd 0.4.4.5 as follows.
1.firewall-cmd --lockdown-on
2.remove all the whitelists, including commands, uids, and contexts
3.firewall-cmd --add-rich-rule='rule XXX'  success, and the rule worked.

I wonder whether firewalld designed like this or not.

**Comment by erig0** [comment]
Yes. This explicitly permitted by default. `/etc/firewalld/lockdown-whitelist.xml` allows root user to make changes even during lockdown.

**Comment by winnermy** [comment]
Maybe I didn't explain the issue clearly.
I removed root user , uid 0 , firewall-config command and all the SELinux context , so there is nothing in /etc/firewalld/lockdown-whitelist.xml. Restart firewalld daemon, and I cannot add service , port ... to a zone, but still can add rich rules

**Comment by erig0** [comment]
Thanks for the clarification. That sounds like a violation. Re-opening so someone can take a look.
